Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py

```diff
@@ -305,15 +302,21 @@ def _get_self_deployed_token(self,
             return self._get_token_device_code(refresh_token, scope, integration_context)
         else:
             # by default, grant_type is CLIENT_CREDENTIALS
+            if self.multi_resource:
+                expires_in = -1  # init variable as an int
+                for resource in self.resources:
+                    self.resource = resource
+                    access_token, expires_in, refresh_token = self._get_self_deployed_token_client_credentials()
+                    self.resource_to_access_token[resource] = access_token
+                self.resource = ''
```

Comment:
Since the authorization in the integration is client credentials grant type and the integration uses more than one resource, we had to add the functionality of multi resource for client credentials in MicrosoftApiModule.py. 
The logic we implemented there is very similar to the logic that is used in the OPROXY auth type (when handling multi resources). 
I agree that it can be improved.

Reply:
Lets improve it. this is not a good practice.
in the OProxy type, we do not override class attribute but implementing it a bit differently in the function.